core/src/main/scala/akka/persistence/r2dbc/journal/JournalDao.scala

```diff
@@ -125,18 +146,29 @@ private[r2dbc] class JournalDao(journalSettings: R2dbcSettings, connectionFactor
             .bindNull(11, classOf[Array[Byte]])
       }
 
+      val p =
+        if (useTimestampFromDb) {
+          12
+        } else {
+          stmt.bind(12, write.dbTimestamp)
+          13
+        }
+
       stmt
-        .bind(12, slice)
-        .bind(13, entityType)
-        .bind(14, write.persistenceId)
-        .bind(15, previousSeqNr)
+        .bind(p, slice)
+        .bind(p + 1, entityType)
+        .bind(p + 2, write.persistenceId)
+        .bind(p + 3, previousSeqNr)
```

Comment:
I think it will be easier to follow this if we just repeat the bindings inside the `if/else`.
Like we did about when matching the option in `write.metadata`.

```scala
if (useTimestampFromDb)
 stmt
    .bind(12, slice)
    .bind(13, entityType)
    .bind(14, write.persistenceId)
    .bind(15, previousSeqNr)
else
  stmt
    .bind(12, write.dbTimestamp)
    .bind(13, slice)
    .bind(14, entityType)
    .bind(15, write.persistenceId)
    .bind(16, previousSeqNr)

```

Reply:
agreed, let me change this